engine/uState.ml

```diff
@@ -261,11 +261,13 @@ let process_universe_constraints ctx cstrs =
                     in
                     LSet.fold fold levels local
               else
-                match Univ.Universe.level l with
-                | Some l ->
-                  Univ.Constraint.add (l, Le, r') local
-                | None ->
-                  if UGraph.check_leq univs l r then local else enforce_leq l r local
+                let fold accu (l', n) = match n with
+                | 0 -> Univ.Constraint.add (l', Le, r') accu
+                | 1 -> Univ.Constraint.add (l', Lt, r') accu
+                | n -> raise (UniverseInconsistency (Le, l, r, None))
+                in
+                let l = Univ.Universe.map (fun p -> p) l in
+                List.fold_left fold local l
```

Comment:
Can't we use enforce_leq instead?

Reply:
`enforce_leq` is doing strange magic in the case `n > 1`. Is that doing the same thing in the end?

Reply:
If `l' <> r' ` it will anomaly instead of your universe inconsistency. Anomaly is probably the right thing to do though.

Reply:
OK, I'll tweak the patch.